eugene-okulik/summerduck
# Lesson 15

# Key Features
Student Management: Create, retrieve, and assign students to groups.
Group Management: Create and assign groups to students.
Book Management: Create books and assign them to students.
Subject and Lesson Management: Create subjects and lessons, and assign marks to students.
Data Validation and Sanitization: Ensures all inputs are validated and sanitized.
Logging: Comprehensive logging for debugging and monitoring.
Error Handling: Robust error handling using decorators.

# Files Added/Modified
database_hv.py: Main script for database operations and business logic.
test_data.py: Generates random test data for students, groups, books, subjects, and lessons.
utils.py: Utility functions for validation, sanitization, and error handling.

# How to Test
Run the main function in database_hv.py
Verify the logs for successful creation and assignment of entities.

<img width="776" alt="Screenshot 2024-10-19 at 6 03 41 PM" src="https://github.com/user-attachments/assets/75059090-b628-4617-9ad0-075216716ccf">

<img width="1357" alt="Screenshot 2024-10-19 at 6 06 26 PM" src="https://github.com/user-attachments/assets/2c796c79-c0a4-4001-b5fd-4f8440becab6">
